src/Simplify.cpp

```diff
@@ -4239,6 +4239,54 @@ class Simplify : public IRMutator {
             } else {
                 expr = op;
             }
+        } else if (op->is_intrinsic(Call::prefetch)) {
+            // Collapse the prefetched region into lower dimension whenever is possible.
+            // TODO(psuriana): Deal with negative strides and overlaps.
+
+            vector<Expr> args(op->args);
+            bool changed = false;
+            for (size_t i = 0; i < op->args.size(); ++i) {
+                args[i] = mutate(op->args[i]);
+                if (!args[i].same_as(op->args[i])) {
+                    changed = true;
+                }
+            }
+
+            // The {extent, stride} args in the prefetch call are sorted
+            // based on the storage dimension in ascending order (i.e. innermost
+            // first and outermost last), so, it is enough to check for the upper
+            // triangular pairs to see if any contiguous addresses exist.
+            for (size_t i = 1; i < args.size(); i += 2) {
+                Expr extent_0 = args[i];
+                Expr stride_0 = args[i + 1];
+                for (size_t j = i + 2; j < args.size(); j += 2) {
+                    Expr extent_1 = args[j];
+                    Expr stride_1 = args[j + 1];
+
+                    if (can_prove(extent_0 * stride_0 == stride_1)) {
+                        Expr new_extent = mutate(extent_0 * extent_1);
+                        const IntImm *int_stride_0 = stride_0.as<IntImm>();
+                        const IntImm *int_stride_1 = stride_1.as<IntImm>();
+                        Expr new_stride = 1;
+                        if (int_stride_0 && int_stride_1) {
```

Comment:
Shouldn't this just always be setting new_stride to stride_0, regardless of whether the strides are IntImm or not?

I think there's a bug here, if stride_0 and stride_1 are IntImm, the new_stride might be larger than one, which will change the region prefetched significantly.